### PR TITLE
[ENH,REF]  Unequal length collection transforms

### PR DESCRIPTION
**Pull request by MatthewMiddlehurst**
- deprecate classes for resizing, padding and truncating and move them to their own unequal length subpackage
- unequal length support for periodogram transform
- adds new functionality to the transforms i.e. padding/truncating/resizing to the min/max series length seen in fit as well as a set value, adding noise for padding, option error throwing for too long/short series etc.
- expands testing for the 3 unequal length removers

### CODE REVIEW

File: aeon/transformations/collection/unequal_length/_resize.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    _get_max_length,
+    _get_min_length,
+)
+
+
+class Resizer(BaseCollectionTransformer):
+    """Resize unequal length time series to equal, fixed length.
+
+    Resize the series using linear interpolation to either a fixed length or
+    finds the max/min length series across all series and channels and resuizes
```
Comment (TonyBagnall) [inline aeon/transformations/collection/unequal_length/_resize.py]: typo

File: aeon/transformations/collection/_pad.py [hunk 2]
```diff
@@ -17,6 +18,13 @@ def _get_max_length(X):
     return max_length
 
 
+# TODO: Remove in v1.3.0
+@deprecated(
+    version="1.2.0",
+    reason="Padder is deprecated and will be moving to . "
```
Comment (TonyBagnall) [inline aeon/transformations/collection/_pad.py]: no full stop after moving to

File: aeon/transformations/collection/unequal_length/_pad.py [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+    >>> X2 = padder.fit_transform(X)
+    >>> X2.shape
+    (10, 4, 200)
+    """
+
+    _tags = {
+        "X_inner_type": ["np-list", "numpy3D"],
+        "capability:multivariate": True,
+        "capability:unequal_length": True,
+        "removes_unequal_length": True,
```
Comment (TonyBagnall) [inline aeon/transformations/collection/unequal_length/_pad.py:67]: this is not now always true, since error_on_long can be false and padded_length can be the minimum in fit

File: aeon/transformations/collection/unequal_length/_pad.py [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+class Padder(BaseCollectionTransformer):
+    """Pad unequal length time series to equal, fixed length.
+
+    Pads the input dataset to either a fixed length or finds the max/min length
+    series across all series and channels and pads to that with zeroes.
+
+    Parameters
+    ----------
+    padded_length  : int, "min" or "max", default="min"
+        Length to pad the series to. If "min", will pad to the shortest
```
Comment (TonyBagnall) [inline aeon/transformations/collection/unequal_length/_pad.py]: will pad the series passed in transform to the shortest

File: aeon/transformations/collection/unequal_length/_commons.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
+"""Common functions for unequal length transformations.
+
+These should ideally be incorporated into the collection data utilities in utils/ in
+the future.
+"""
+
+
+def _get_min_length(X):
+    min_length = X[0].shape[1]
+    for x in X:
```
Comment (TonyBagnall) [inline aeon/transformations/collection/unequal_length/_commons.py]: cant you do this without the loop? somerthing like

def _get_max_length(X):
    if isinstance(X, np.ndarray) and X.ndim == 3:
        return X.shape[2]
    else:
        lengths = [x.shape[1] for x in X]
        return max(lengths)

File: aeon/transformations/collection/unequal_length/_pad.py (hunk 3 above)
Comment (MatthewMiddlehurst) [reply aeon/transformations/collection/unequal_length/_pad.py:67]: good catch, made it dynamic based on error_on_long

File: aeon/transformations/collection/unequal_length/_commons.py (hunk 5 above)
Comment (MatthewMiddlehurst) [reply aeon/transformations/collection/unequal_length/_commons.py]: Done

**Review by TonyBagnall — APPROVED** [review]
LGTM
